rocAL_pybind/amd/rocal/pipeline.py

```diff
@@ -153,22 +152,15 @@ def copyToExternalTensor(self, array,  multiplier, offset, reverse_channels, ten
         b.rocalToTensor(self._handle, ctypes.c_void_p(array.data_ptr()), tensor_format, tensor_dtype,
                         multiplier[0], multiplier[1], multiplier[2], offset[0], offset[1], offset[2], (1 if reverse_channels else 0), self._output_memory_type, max_roi_height, max_roi_width)
 
-    """
-    NOTE: getOneHotEncodedLabels is not defined. This function does not have a cpp equivalent. Bring back once cupy/fucntion is fixed
-    def get_one_hot_encoded_labels(self, array, device):
+    def get_one_hot_encoded_labels(self, array_ptr, device):
         if device == "cpu":
-            if (isinstance(array, np.ndarray)):
-                b.getOneHotEncodedLabels(self._handle, array.ctypes.data_as(
-                    ctypes.c_void_p), self._num_classes, 0)
-            else:  # torch tensor
-                return b.getOneHotEncodedLabels(self._handle, ctypes.c_void_p(array.data_ptr()), self._num_classes, 0)
+            b.getOneHotEncodedLabels(self._handle, array_ptr, self._num_classes, types.HOST_MEMORY)
```

Comment:
the same function is called with same parameters for both cpu and gpu. In this case, if else condition can be removed

Reply:
Merged CPU and GPU functionalities for one hot labels